src/pages/index.js

```diff
@@ -16,33 +18,65 @@ const profileNameInput = document.getElementById("profile-name-input");
 
 profileEditButton.addEventListener("click", () => {
   const userInfo = profileUserInfo.getUserInfo();
-  profileDescriptionInput.value = userInfo.description;
   profileNameInput.value = userInfo.name;
+  profileDescriptionInput.value = userInfo.description;
   profileModal.open();
 });
 
 addCardButton.addEventListener("click", () => {
   newCardModal.open();
 });
 
+const deleteModal = new PopupConfirm({ popupSelector: "#are-you-sure" });
+
+deleteModal.setEventListeners();
+
 const newCardModal = new PopupWithForm(
   "#profile-card-modal",
   handleCardContent,
-  "#card-add-button"
+  "#card-add-button" // TODO do we need this third argument?
 );
 
 newCardModal.setEventListeners();
+
 const profileName = document.querySelector(".profile__name");
 const profileDescription = document.querySelector(".profile__description");
-const profileUserInfo = new UserInfo({ profileName, profileDescription });
+const profileAvatar = document.querySelector(".profile__image");
+const avatarEditBtn = document.querySelector(".pencil-icon");
+const profileUserInfo = new UserInfo({
+  profileName,
+  profileDescription,
+  profileAvatar,
+});
 
 const profileModal = new PopupWithForm("#profile-edit-modal", (userInput) => {
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```